### ISSUE TITLE
Back arrow after reload breaks navigation

### ISSUE THREAD

**Issue by KucharczykL** [jellyfin/jellyfin-web#1848]
**Describe The Bug**
Clicking back in UI after page reload breaks navigation/website.

**Steps To Reproduce**

1. Go to a TV show/movie detail (library -> TV/movie name)
2. Reload website either by clicking the reload button or pressing F5.
3. Click the back arrow in the Jellyfin UI.
4. The address changes to https://domain/web#!/home.html and clicking back or clicking home doesn't do anything. Clicking library in the side bar gives an error: "Error processing request." 

**Expected Behavior**
The browser goes back in history after clicking back.

**Logs**
```
[16:38:49] [ERR] [145] Emby.Server.Implementations.HttpServer.HttpListenerHost: Could not find handler for /controllers/home.js
[16:38:49] [ERR] [145] Emby.Server.Implementations.HttpServer.HttpListenerHost: Error processing request: Unable to find the specified file. URL: http://domain/controllers/home.js?v=27
[16:38:52] [INF] [84] Emby.Server.Implementations.HttpServer.HttpListenerHost: WS ::ffff:172.17.0.1 closed
[16:38:52] [ERR] [136] Emby.Server.Implementations.HttpServer.HttpListenerHost: Could not find handler for /tv.html
[16:38:52] [ERR] [136] Emby.Server.Implementations.HttpServer.HttpListenerHost: Error processing request: Unable to find the specified file. URL: http://domain/tv.html?topParentId=3bd70d37ca5b48bad9652bf3c4562b02
```

**Screenshots**

**System (please complete the following information):**
 - Linux
 - Browser: Firefox 80.0
 - Jellyfin Version: 10.6.3

**Additional Context**
I am running Jellyfin locally on a server in Docker container behind NGINX reverse proxy with custom domain (via NGINX Proxy Manager). Setup is according to the Jellyfin wiki.

**Comment by ThibaultNocchi** [comment]
I can confirm it works without a reverse proxy. Then this may be caused by server side management of delivering the content.

**Comment by KucharczykL** [comment]
I have tried it just now without the reverse proxy (just bare local IP) and it doesn't work. Tried in Firefox and Chromium. What else can I try?

**Comment by ThibaultNocchi** [comment]
My bad, I'm stupid. I clicked the FF back button. It sure doesn't work anywhere then haha.

**Comment by KucharczykL** [comment]
It's also my bad, I should have specified which back arrow. Fixed the 3rd step.

**Comment by ThibaultNocchi** [comment]
No problem, and you already specified it was in the UI in your first line. I should have read better.

So this bug is due to how the web client is developed as an SPA but with old and hacky techs. There's a rewrite in the plan for the whole routing part, it should fix those kind of errors. Meanwhile, I'm not experienced enough to further help you for now.

**Comment by ThinkChaos** [comment]
This bug is the most annoying part of my Jellyfin experience (which is otherwise great) and I'd like to help solve it.
Any info on the rewrite, or tips on what the "hacky techs" that would need fixing are?

**Comment by thornbill** [comment]
I believe this is fixed in unstable now (will be in 10.9 not 10.8).

**Comment by ThinkChaos** [comment]
Thanks for the quick reply, I tested a recent nightly today and it does indeed work as expected for me!
Looking forward to 10.9.

**Comment by obiwantoby** [comment]
Is there an approved way of opening up this issue? All these duplicates describe the exact behavior on the latest build as of 1/25/23.

**Comment by kimboslice99** [comment]
@obiwantoby read the comments. will be fixed in 10.9
